### PR TITLE
multi_thread_blinky sample for 96b_carbon

### PR DESCRIPTION
**Pull request by ric96**
Signed-off-by: Sahaj Sarup <sahaj.sarup@linaro.org>

### CODE REVIEW

**Comment by pfalcon** [comment]
What's so special about 96b_carbon that this sample is specifically for it? Why do we need yet another blink sample in the first place?

**Review by galak — CHANGES_REQUESTED** [review]
commit needs a commit message.

**Comment by ric96** [comment]
> Why do we need yet another blink sample in the first place?
Blink is not the intended demo. I intended this to be a demo that shows the simplest way to spawn threads.

> What's so special about 96b_carbon that this sample is specifically for it?
Agreed. Would you consider it as a separate demo?

File: samples/boards/96b_carbon/multi_thread_blinky/README.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+Overview
+********
+
+This example demonstrates spawning of multiple threads using K_THREAD_DEFINE.
+
+The example works by spawing three threads each to controll a separate LED 
+on the 96Boards Carbon Board. Each of these LEDs (USR1, USR2 and BT) have 
+their individual loop control and timing logic defind inside separate functions.
+
+blink1() controls the USR1 LED that has a 100ms sleep cycle
```
Comment (dbkinder) [inline samples/boards/96b_carbon/multi_thread_blinky/README.rst]: Let's make this a bullet list by adding a ``-`` before each new item, and from the code it looks like blink3 never makes it to 600ms so:

     - blink1() controls the USR1 LED with a 100ms on/off blink rate
     - blink2() controls the USR2 LED with a 1000ms (1 sec) on/off blink rate
     - blink3() controls the BT LED with a 100ms on/off blink rate that
       increases by 100ms up to 500ms, then resets back to 100ms

File: samples/boards/96b_carbon/multi_thread_blinky/README.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+blink2() controls the USR2 LED that has a 1000ms sleep cycle
+blink3() controls the BT LED that increases sleep by 100ms every cycle
+	 upto 600ms and then resets to 100ms
+
+
+Each thread is then spawned during compile time using K_THREAD_DEFINE
```
Comment (dbkinder) [inline samples/boards/96b_carbon/multi_thread_blinky/README.rst]: I think technically, the thread can't be "spawned during compile time", so maybe change to:

     Each thread is then defined at compile time using K_THREAD_DEFINE.

File: samples/boards/96b_carbon/multi_thread_blinky/README.rst [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+Apart from that, a main() thread is also present that provides a basic shell
+from the shell sample at <zephyr_base>/samples/subsys/shell/shell/
```
Comment (dbkinder) [inline samples/boards/96b_carbon/multi_thread_blinky/README.rst]: s/from/based on/

File: samples/boards/96b_carbon/multi_thread_blinky/README.rst [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+Building
+********
+
+.. code-block:: console
+    $ cd samples/boards/96b_carbon/multi_thread_blinky
```
Comment (dbkinder) [inline samples/boards/96b_carbon/multi_thread_blinky/README.rst]: need a blank line before the actual code block contents

File: samples/boards/96b_carbon/multi_thread_blinky/README.rst [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    $ make
+
+Flashing via OTG
+    $ sudo dfu-util -d [0483:df11] -a 0 -D outdir/96b_carbon/zephyr.bin -s 0x08000000
```
Comment (dbkinder) [inline samples/boards/96b_carbon/multi_thread_blinky/README.rst]: add the ``.. code-block`` directive before this code (and a blank line :)

File: samples/boards/96b_carbon/multi_thread_blinky/README.rst [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+Flash via UART
+    $ sudo stm32flash -b 115200 -g 0x0 -w outdir/96b_carbon/zephyr.bin /dev/ttyUSB0
```
Comment (dbkinder) [inline samples/boards/96b_carbon/multi_thread_blinky/README.rst]: add the ``.. code-block`` directive before this code (and a blank line :)

**Review by dbkinder — CHANGES_REQUESTED** [review]
suggested edits to fix doc build errors, and spelling/clarity

**Review by pfalcon — CHANGES_REQUESTED** [review]
> Blink is not the intended demo. I intended this to be a demo that shows the simplest way to spawn threads.

I believe that the canonical threading demo is "philosophers".

If you really want to do something even simpler, sounds good, but please make it a generic, not board-specific sample (which may require limiting to 2 LEDs, maybe make 3rd thread output something on uart).

And if the aim is to make this the simplest threading demo, please make it such, for example, why does it define a shell command?

**Comment by ric96** [comment]
@pfalcon 
> If you really want to do something even simpler, sounds good, but please make it a generic, not board-specific sample (which may require limiting to 2 LEDs, maybe make 3rd thread output something on uart).
Ok. got it. I'll close this one and add a new generic one soon just so that the PR title isn't confusing.
thanks :)

**Pull request closed without merging**